### PR TITLE
feat: stargate: grpc client refactoring and add grpc query function

### PR DESCRIPTION
**Pull request by gyuguen**
Refactoring and GRPC query function
- remove unused resource file in  build.gradle setting
- commons-lang3 version change(3.6 -> 3.12.0)
- GrpcStub colleted to ``GrpcStub.java`
- Add query in PanaceaGrpcClient
  - topic, didDocumentWithSeq, txResponse(s), block
-  Change fee Coin & gasLimit -> Fee class

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -17,21 +17,10 @@ plugins {
...
         }
     }
-    test {
-        proto {
-            srcDir 'src/test/resources/proto'
-        }
-        proto {
-            srcDir 'src/test/java'
-        }
-    }
```
Comment (gyuguen) [inline build.gradle:34]: Apply before PR.
Remove unused resource

File: src/main/java/org/medibloc/panacea/GrpcStub.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package org.medibloc.panacea;
```
Comment (gyuguen) [inline src/main/java/org/medibloc/panacea/GrpcStub.java:1]: I gathered the `stub`s

File: src/main/java/org/medibloc/panacea/PanaceaGrpcClient.java [hunk 3]
```diff
@@ -43,24 +38,68 @@ public BaseAccount getAccount(String address) throws PanaceaApiException {
...
 
+    public TxResponse getTxResponse(String txHash) {
+        GetTxRequest request = GetTxRequest.newBuilder()
+                .setHash(txHash)
+                .build();
+        GetTxResponse response = grpcStub.getTxServiceStub().getTx(request);
+        return response.getTxResponse();
+    }
+
+    public List<TxResponse> getTxResponsesByHeight(long height) {
```
Comment (gyuguen) [inline src/main/java/org/medibloc/panacea/PanaceaGrpcClient.java:85]: Before spec. Get txResponses required to tx.height.
So I implement it as it is

File: src/main/java/org/medibloc/panacea/domain/DIDs.java [hunk 4]
```diff
@@ -24,11 +24,9 @@ private static String encodePubKey(byte[] pubKey) throws NoSuchAlgorithmExceptio
     }
 
 
-    public static VerificationMethod createVerificationMethod(DIDWallet wallet, String key) throws NoSuchAlgorithmException {
+    public static VerificationMethod createVerificationMethod(String did, DIDWallet wallet, String key) {
```
Comment (gyuguen) [inline src/main/java/org/medibloc/panacea/domain/DIDs.java:27]: I changed the did value to be taken from outside.
Because verificationMethodId is similar to did.
If you create multiple verificationMoethods, verificationMethodId is affected to did.

Comment (gyuguen) [reply src/main/java/org/medibloc/panacea/domain/DIDs.java:27]: I think priv/pub key can always be different, verificationMethodId is similar to did.

**Comment by gyuguen** [comment]
@youngjoon-lee 
I want to hear your opinion.
Request to tx, the sequence of the account that execute it is changed.
Wallets that run tx once cannot be re-run.(Because of increased sequence).
Is the `ensureWalletIsReady` intended to be used only when get initial wallet?

```java
    public synchronized void ensureWalletIsReady(PanaceaGrpcClient client) throws PanaceaApiException {
        if (accountNumber == null) {
            initAccount(client);
        }
        if (sequence == null) {
            reloadAccount(client);
        }

        if (chainId == null) {
            initChainId(client);
        }
    }
```

File: src/main/java/org/medibloc/panacea/domain/DIDs.java (hunk 4 above)
Comment (youngjoon-lee) [reply src/main/java/org/medibloc/panacea/domain/DIDs.java:27]: Yes. That's true :+1:

**Comment by youngjoon-lee** [comment]
@gyuguen Yes. Whenever you execute a tx, you have to set a exact sequence of the account. 

In the blockchain, the sequence is increased after a tx is completed.
So, the user must call [wallet.increaseAccountSequence()](https://github.com/medibloc/panacea-java/blob/884386d9453680d4edafa269aa35d0471611c7a5/src/test/java/org/medibloc/panacea/PanaceaApiRestClientTest.java#L85) to increase the `sequence` field in the `Wallet` object. It doesn't communicate with the chain.

Another way, the user can create a new `Wallet` object and call `wallet.ensureWalletIsReady(client)` whenever he wants to execute a tx.
Then, it will fetch the latest sequence of the account from the chain. 
In other words, `wallet.ensureWalletIsReady(client)` should be called once after the `Wallet` object is created.

In both ways, transactions can be failed if there are more than one executor that uses the same account. They will try to excute a tx with the same account number and sequence, but only one of them will be successful. It's the design choice of the cosmos blockchain in order to prevent replay attacks.

File: src/main/java/org/medibloc/panacea/PanaceaGrpcClient.java (hunk 3 above)
Comment (youngjoon-lee) [reply src/main/java/org/medibloc/panacea/PanaceaGrpcClient.java:85]: Aha. Yeah. I'm not sure whether SDK users are actually using this features :), but it's good to have for backward-compatibility because it's simple.

**Review by youngjoon-lee — APPROVED** [review]
Awesome!

**Comment by gyuguen** [comment]
> @gyuguen Yes. Whenever you execute a tx, you have to set a exact sequence of the account.
> 
> In the blockchain, the sequence is increased after a tx is completed.
> So, the user must call [wallet.increaseAccountSequence()](https://github.com/medibloc/panacea-java/blob/884386d9453680d4edafa269aa35d0471611c7a5/src/test/java/org/medibloc/panacea/PanaceaApiRestClientTest.java#L85) to increase the `sequence` field in the `Wallet` object. It doesn't communicate with the chain.
> 
> Another way, the user can create a new `Wallet` object and call `wallet.ensureWalletIsReady(client)` whenever he wants to execute a tx.
> Then, it will fetch the latest sequence of the account from the chain.
> In other words, `wallet.ensureWalletIsReady(client)` should be called once after the `Wallet` object is created.
> 
> In both ways, transactions can be failed if there are more than one executor that uses the same account. They will try to excute a tx with the same account number and sequence, but only one of them will be successful. It's the design choice of the cosmos blockchain in order to prevent replay attacks.

OK. Thank you!
